float32 angleA
float32 angleB
float32 angleC
float32 angleD